drivers/i2s/i2s_ll_stm32.c

```diff
@@ -182,7 +182,8 @@ static int i2s_stm32_configure(const struct device *dev, enum i2s_dir dir,
 	 * When I2S data format is selected parameter channels is ignored,
 	 * number of words in a frame is always 2.
 	 */
-	const uint32_t num_channels = i2s_cfg->format & I2S_FMT_DATA_FORMAT_MASK
+	const uint32_t num_channels = ((i2s_cfg->format & I2S_FMT_DATA_FORMAT_MASK)	==
+				      I2S_FMT_DATA_FORMAT_I2S)
```

Comment:
```suggestion
	const uint32_t num_channels = ((i2s_cfg->format & I2S_FMT_DATA_FORMAT_MASK) ==
				      I2S_FMT_DATA_FORMAT_I2S) ? 2U : i2s_cfg->channels;
```

Reply:
@erwango I have addressed most of these issues for @cdesjardins, my colleage ;)